pipeline_dp/dp_computations.py

```diff
@@ -505,7 +513,13 @@ def create_from_std_deviation(cls, normalized_stddev: float,
                 stddev),
             l2_sensitivity=l2_sensitivity)
 
-    def add_noise(self, value: Union[int, float]) -> float:
+    def add_noise(
+            self, value: Union[int, float,
+                               np.ndarray]) -> Union[float, np.ndarray]:
+        if isinstance(value, np.ndarray):
+            add_noise_vectorized = np.vectorize(
+                lambda x: self._mechanism.add_noise(1.0 * x))
+            return add_noise_vectorized(value)
         return self._mechanism.add_noise(1.0 * value)
```

Comment:
this is duplicate can code, does it make sense to define it in the parent abstract code?

also at line 523 nit: (same above for laplace) we can reuse lambda from numpy case and call it here